doc/apiary.apib

```diff
@@ -0,0 +1,754 @@
+                            "created": "2017-12-20T18:42:21.091596+00:00",
+                            "id": 15,
+                            "label": "led",
+                            "template_id": "4",
+                            "type": "dynamic",
+                            "value_type": "bool"
+                        },
+                        {
+                            "created": "2017-12-20T18:42:21.092641+00:00",
+                            "id": 16,
+                            "label": "fan",
+                            "template_id": "4",
+                            "type": "dynamic",
+                            "value_type": "bool"
+                        }
+                    ],
+                    "created": "2017-12-20T18:42:21.087782+00:00",
+                    "id": 4,
+                    "label": "SensorModel"
+                },
+                "result": "ok"
+            }
+
++ Response 404 (application/json)
+
+            {
+                "message": "No such template: 123",
+                "status": 404
+            }
+
+
+### Get template info [GET]
+Retrieves all information from a specific template
++ Response 200 (application/json)
+
+            {
+                "attrs": [
+                    {
+                        "created": "2017-12-20T18:42:21.089367+00:00",
+                        "id": 13,
+                        "label": "temperature",
+                        "template_id": "4",
+                        "type": "dynamic",
+                        "value_type": "float"
+                    },
+                    {
+                        "created": "2017-12-20T18:42:21.090601+00:00",
+                        "id": 14,
+                        "label": "pressure",
+                        "template_id": "4",
+                        "type": "dynamic",
+                        "value_type": "float"
+                    },
+                    {
+                        "created": "2017-12-20T18:42:21.091596+00:00",
+                        "id": 15,
+                        "label": "led",
+                        "template_id": "4",
+                        "type": "dynamic",
+                        "value_type": "bool"
+                    },
+                    {
+                        "created": "2017-12-20T18:42:21.092641+00:00",
+                        "id": 16,
+                        "label": "fan",
+                        "template_id": "4",
+                        "type": "dynamic",
+                        "value_type": "bool"
+                    }
+                ],
+                "created": "2017-12-20T18:42:21.087782+00:00",
+                "id": 4,
+                "label": "SensorModel"
+            }
+    
++ Response 404 (application/json)
+
+            {
+                "message": "No such template: 123",
+                "status": 404
+            }
+
+### Update template info [PUT]
+Replaces all attributes from a specific template.
```

Comment:
Whats the use of the attribute Id if we always have to replace all of then? What happens if we do an update with the same attributes? Are they replaced and the history is lost or are they kept the way they were?

Reply:
I'll confirm that, but they are not lost. You can still read the history data with the same labels.